api/scpca_portal/models/original_file.py

```diff
@@ -140,69 +154,85 @@ def purge_deleted_files(
         deletable_files.delete()
         return deletable_file_list
 
-    @staticmethod
-    def is_project_file(s3_key: Path) -> bool:
-        """Checks to see if file is a project data file, and not a library data file."""
-        # project files will not have sample subdirectories
-        return next((True for p in s3_key.parts if common.SAMPLE_ID_PREFIX in p), False)
 
-    @staticmethod
-    def get_relationship_ids(s3_key: Path) -> Tuple:
-        """Parses s3_key and returns project, sample and library ids."""
-        project_id = next(
-            (p for p in s3_key.parts if common.PROJECT_ID_PREFIX in p),
-            None,  # the only file w.o. a project id path part should be the projects metadata file
+@dataclass
+class S3KeyInfo:
+    s3_key: Path
+    project_id_part: str | None
+    sample_id_part: str | None
+    library_id_part: str | None
+    is_merged: bool
+    is_bulk: bool
+
+    # is_spatial:
+    def __init__(self, s3_key: Path):
+        self.s3_key = s3_key
+        self.project_id_part = utils.find_matching_part(common.PROJECT_ID_PREFIX, s3_key.parts)
+        self.sample_id_part = utils.find_matching_part(common.SAMPLE_ID_PREFIX, s3_key.parts)
+        self.library_id_part = utils.find_matching_part(common.LIBRARY_ID_PREFIX, s3_key.parts)
+        self.is_merged = "merged" in s3_key.parts
+        self.is_bulk = "bulk" in s3_key.parts
+
+    @property
+    def project_id(self):
+        return self.project_id_part
+
+    @property
+    def sample_id(self):
+        return self.sample_id_part
+
+    @property
+    def library_id(self):
+        if self.library_id_part:
+            return self.library_id_part.split("_")[0]
+        return self.library_id_part
+
+    @property
+    def is_project_file(self):
+        """Project files have project dirs but don't have sample dirs"""
+        return not self.sample_id_part
+
+    @property
+    def is_spatial(self):
+        if self.is_project_file:
+            return False
+
+        # all spatial files have "spatial" appended to the libary part of their file path
+        return self.library_id_part.endswith("spatial")
+
+    @property
+    def is_single_cell(self):
+        if self.is_project_file:
+            return False
+        # spatial and single_cell are mutually exclusive
+        return not self.is_spatial
+
+    @property
+    def is_cite_seq(self):
+        if self.is_project_file:
+            return False
+        return common.CITE_SEQ_FILE_SUFFIX in self.s3_key.name
```

Comment:
```suggestion
        return self.s3_key.name.endswith(common.CITE_SEQ_FILENAME_ENDING)
        
        # where CITE_SEQ_FILENAME_ENDING = "_adt.h5ad"
```
Since we know that Cite-Seq files are only for anndata files we can assert this instead. Would also make sense to reference the FORMAT_EXTENSIONS["ANN_DATA"] as this will change if we update the AnnData extension. (we have before)